analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTest.java

```diff
@@ -489,23 +492,115 @@ public void analyze_appActivityCompletelyPrecedingLastCachedActivity_existingAct
+				a -> a.getStartTime().equals(startTime.toLocalDateTime()) && a.getEndTime().equals(endTime.toLocalDateTime()))
+				.findFirst();
+		assertTrue("Expect that new activity is added", addedActivity.isPresent());
+	}
+
+	@Test
+	public void analyze_appActivityCompletelyPrecedingLastCachedActivityOverlappingExistingActivity_existingActivityUpdatedButNoNewGoalConflictMessagesCreated()
+	{
+		ZonedDateTime now = now();
+		ZonedDateTime existingActivityTimeStartTime = now.minusMinutes(20);
+		ZonedDateTime existingActivityTimeEndTime = existingActivityTimeStartTime.plusMinutes(10);
+
+		Activity existingActivityOne = createActivity(existingActivityTimeStartTime, existingActivityTimeEndTime, "Poker App");
+		Activity existingActivityTwo = createActivity(now, now, "Poker App");
+		mockExistingActivities(gamblingGoal, existingActivityOne, existingActivityTwo);
+
+		when(mockActivityRepository.findOverlapping(any(DayActivity.class), any(UUID.class), any(String.class),
+				any(LocalDateTime.class), any(LocalDateTime.class))).thenAnswer(new Answer<Activity>() {
+					@Override
+					public Activity answer(InvocationOnMock invocation) throws Throwable
+					{
+						return existingActivityOne;
+					}
+				});
+
+		// Test an activity
+		ZonedDateTime startTime = existingActivityTimeStartTime.plusMinutes(5);
+		ZonedDateTime endTime = startTime.plusMinutes(7);
+
+		service.analyze(userAnonId, createSingleAppActivity("Poker App", startTime, endTime));
+
+		verifyNoGoalConflictMessagesCreated();
+
+		// Verify that a database lookup was done finding the existing DayActivity to update
+		verify(mockDayActivityRepository, atLeastOnce()).findOne(userAnonId, now.toLocalDate(), gamblingGoal.getId());
+
+		// Verify that the day was updated
+		verify(mockUserAnonymizedService, atLeastOnce()).updateUserAnonymized(userAnonEntity);
+		List<WeekActivity> weekActivities = gamblingGoal.getWeekActivities();
+		assertThat("One week activity created", weekActivities.size(), equalTo(1));
+		List<DayActivity> dayActivities = weekActivities.get(0).getDayActivities();
+		assertThat("One day activity created", dayActivities.size(), equalTo(1));
+		DayActivity dayActivity = dayActivities.get(0);
+
+		// Verify that the activity was not updated. An existing old activity should be updated, not the cache
+		verify(mockAnalysisEngineCacheService, never()).updateLastActivityForUser(eq(userAnonId), eq(gamblingGoal.getId()),
+				any());
+
+		assertThat("Expect last activity unchanged", dayActivity.getLastActivity(), equalTo(existingActivityTwo));
+		assertThat("Expect right activity start time", existingActivityOne.getStartTime(),
+				equalTo(existingActivityTimeStartTime.toLocalDateTime()));
+		assertThat("Expect right activity end time", existingActivityOne.getEndTime(), equalTo(endTime.toLocalDateTime()));
+	}
+
+	@Test
+	public void analyze_appActivityOverlappingLastCachedActivityBeginAndEnd_existingActivityUpdatedButNoNewGoalConflictMessagesCreated()
```

Comment:
Suggestion: `..._lastCachedActivityUpdated...` in contrast with the previous one

Reply:
Are you sure the last cached activity is updated here? I can't see the source on my tablet here, but if you're sure, renaming is fine.